### DISCUSSION TITLE
Custom messages and services types

### DISCUSSION THREAD

**Discussion by BielLopes** [larics/icuas22_competition#36] [Q&A]
As stated in [1](https://github.com/larics/icuas22_competition/discussions/8#discussioncomment-2182587), the organizers do not intend to release the implementation of custom messages. I would like to argue against this decision and ask for a review of this condition. To make the code more eligible and modular. For example, creating a specific service that returns which of the walls and which position on the wall contains the AR tag, the best way to work as a team and make a good software architecture.
Can this decision be reviewed?

**Comment by fpetric** [comment]
Hi @BielLopes,

everything is reviewable :) and we tend to agree with your statement :) that part was probably poorly written by me, you are allowed to use services and messages based on built-in ROS message types to communicate between nodes.
